divvy/cloudfilebrowser.js

```diff
@@ -911,35 +975,52 @@ var cloudFileBrowser = (function() {
             }, callbackArgs);
         },
 
-        drawEl: function(data, element, path) {
+        drawEl: function(data, element, path, keywoard) {
             // Clean up load screen
             cloudFileBrowser.hideLoading();
 
 
             $('div.' + element + ' .listTable, div.' + element + ' .breadcrumb').remove();
 
+            // remove search wrapper
+            $('.search-wrapper').remove();
+
             // Call for table from helper class
             var tableHTML = this.buildTable(data, true, path, element);
 
             // Append data returned and start screen adjustment via CSS3 class
             $(container + ' .' + element).addClass('provisioned').append(tableHTML);
 
+            if(keywoard !== '' && keywoard !== undefined) {
+                this.disableSearchOpen();
+                $("#js-search-box-form-wrapper").show();
+                $('#js-search-box').val(keywoard);
+            }            
+
             this.animateTable(element);
             this.bindFileDragDrop(element, path);
             this.bindAddFiles(element, path);
             this.bindBreadCrumbClick(element);
             this.bindFileInfo(element);
+            this.bindSearchBox(element, path);
         },
 
         buildTable: function(data, isNew, path, element) {
-
             if (isNew == true) {
 
                 var tableHTML = '',
                     trailingpath;
 
                 cloudFileBrowser.selectedFiles[element] = new Array();
 
+                tableHTML += '<div class="search-wrapper">' +
```

Comment:
Doesn't this code make you appreciate Divvy 3.0 frontend? :)

Reply:
Absolutely!